DrawObject[] draw_object_list